core/extraction/function_extractor.py

```
@@ -17,15 +17,11 @@
-class FunctionExtractor:
-    """Handles extraction of functions from Python source code."""
+class BaseExtractor:
+    """Base class for shared extraction methods."""
 
-    def __init__(
-        self,
-        context,
-        correlation_id: Optional[str] = None,
-    ) -> None:
-        """Initialize the function extractor."""
+    def __init__(self, context, correlation_id: Optional[str] = None) -> None:
```

Ruff (0.8.2)

Missing type annotation for function argument `context`

(ANN001)